src/hooks/useENS.ts

```diff
@@ -1,23 +1,40 @@
 import { useEffect, useState } from "react"
+import { ChainId } from "../constants"
 import { ethers } from "ethers"
 import { useActiveWeb3React } from "."
 
 type ReturnType = { ensName: string | null }
 
 export const useENS = (address: string | null | undefined): ReturnType => {
   const [ensName, setENSName] = useState<string | null>(null)
-  const { library } = useActiveWeb3React()
+  const { library, chainId } = useActiveWeb3React()
 
   useEffect(() => {
     async function resolveENS() {
-      if (address && library && ethers.utils.isAddress(address)) {
+      if (
+        address &&
+        library &&
+        ethers.utils.isAddress(address) &&
+        chainId === ChainId.MAINNET
+      ) {
         const provider = new ethers.providers.Web3Provider(library.provider)
         const ensName = await provider.lookupAddress(address)
         if (ensName) setENSName(ensName)
+      } else if (
+        address &&
+        library &&
+        ethers.utils.isAddress(address) &&
+        chainId !== ChainId.MAINNET
+      ) {
+        // Get ens name from main chain when network is non-mainnet
+        const ensNameOnMainnet = await ethers
+          .getDefaultProvider()
```

Comment:
Can we use REACT_APP_NETWORK_URL as an argument here? My concern is that `getDefaultProvider` basically spams free endpoints, meaning that it creates a ton of network requests and will eventually return 429s or something like that. Here's was the network looks like when moving to another chain: 

<img width="449" alt="Screen Shot 2022-01-27 at 7 15 54 PM" src="https://user-images.githubusercontent.com/7607886/151459432-60cbeebf-dd80-4595-aa97-db13ce88dff2.png">

Reply:
You are right.